tests/pytests/test_info.py

```diff
@@ -154,3 +154,41 @@ def test_info_text_tag_overhead(env):
   res = index_info(env, 'idx')
   env.assertEqual(float(res['tag_overhead_sz_mb']), 0)
   env.assertEqual(float(res['text_overhead_sz_mb']), 0)
+
+def test_vecsim_info_stats_memory(env):
+  env = Env(protocol=3)
+  vec_size = 6
+  data_type = 'FLOAT16'
+  env.expect('FT.CREATE', 'idx', 'ON', 'HASH', 'SCHEMA', 'vector', 'VECTOR', 'FLAT', 6, 'DIM', 6, 'TYPE', 'float16', 'DISTANCE_METRIC', 'L2').ok()
+  load_vectors_to_redis(env, 1000, 0, vec_size, data_type)
+  def get_memory(info): return info[info.index('MEMORY') + 1]
+  total_memory = 0
+  for shard_conn in shardsConnections(env):
+    total_memory += get_memory(shard_conn.execute_command(debug_cmd(), 'VECSIM_INFO', 'idx', 'vector'))
+  info = shard_conn.execute_command('ft.info', 'idx')
+  env.assertTrue("field statistics" in info)
+  env.assertAlmostEqual(total_memory, info["field statistics"][0]["memory"], delta=env.shardsCount)
+
+def test_vecsim_info_stats_marked_deleted(env):
+  env = Env(protocol=3, moduleArgs='WORKERS 1')
+  conn = env.getClusterConnectionIfNeeded()
+  vec_size = 6
+  data_type = 'FLOAT16'
+  load_vectors_to_redis(env, 1000, 0, vec_size, data_type)
+  conn.execute_command('FT.CREATE', 'idx', 'ON', 'HASH', 'SCHEMA', 'vector', 'VECTOR', 'HNSW', 6, 'DIM', 6, 'TYPE', 'float16', 'DISTANCE_METRIC', 'L2')
+  for i in range(1, 101):
+    conn.execute_command('DEL', f'doc{i}')
```

Comment:
i don't think you are removing any vectors here (note that the document name used in load_vectors_to_redis is different)
i would compare the expected marked_deleted to the number of deleted docs